pkg/validator/cluster.go

```diff
@@ -465,6 +481,14 @@ func admitClusterCR(
 		patches,
 	)
 
+	patches = addServiceType(&clusterCR, kdConfigCR, patches)
+
+	// If cluster already exists, check for property changes.
+	if ar.Request.Operation == v1beta1.Update {
+		valErrors = validateGeneralChanges(&clusterCR, &prevClusterCR, valErrors)
+		valErrors = validateRoleChanges(&clusterCR, &prevClusterCR, valErrors)
+	}
```

Comment:
I'm pondering the comment that got deleted from this block and how much it still applies.

Perhaps what we should do is accumulate the errors inside this block into a different list (instead of valErrors). Then if there are any errors generated in this block, just overwrite valErrors with them. Something like:
```go
// If cluster already exists, check for property changes.
if ar.Request.Operation == v1beta1.Update {
    var changeErrors []string
    changeErrors = validateGeneralChanges(&clusterCR, &prevClusterCR, changeErrors)
    changeErrors = validateRoleChanges(&clusterCR, &prevClusterCR, changeErrors)
    // If un-change-able properties are being changed, ignore all other error
    // messages in favor of those. (The reason we didn't just do this check
    // first and then skip other validation is because this check depends on
    // the defaulting logic that happens in those other functions.)
    if len(changeErrors) != 0 {
        valErrors = changeErrors
    }
}
```

Thoughts?